src/lib/server/cms.ts

```diff
@@ -59,68 +108,24 @@ export async function cmsFromContent(
 	const tagProductsSlugs = new Set<string>();
 	const gallerySlugs = new Set<string>();
 
-	const tokens: Array<
-		| {
-				type: 'html';
-				raw: string;
-		  }
-		| {
-				type: 'productWidget';
-				slug: string;
-				display: string | undefined;
-				raw: string;
-		  }
-		| {
-				type: 'challengeWidget';
-				slug: string;
-				raw: string;
-		  }
-		| {
-				type: 'sliderWidget';
-				slug: string;
-				autoplay: number | undefined;
-				raw: string;
-		  }
-		| {
-				type: 'tagWidget';
-				slug: string;
-				display: string | undefined;
-				raw: string;
-		  }
-		| {
-				type: 'specificationWidget';
-				slug: string;
-				raw: string;
-		  }
-		| {
-				type: 'pictureWidget';
-				slug: string;
-				raw: string;
-				fit?: 'cover' | 'contain';
-				width?: number;
-				height?: number;
-		  }
-		| { type: 'contactFormWidget'; slug: string; raw: string }
-		| { type: 'countdownWidget'; slug: string; raw: string }
-		| { type: 'tagProducts'; slug: string; display: string | undefined; raw: string }
-		| {
-				type: 'galleryWidget';
-				slug: string;
-				display: string | undefined;
-				raw: string;
-		  }
-	> = [];
+	const tokens: {
+		desktop: Array<TokenObject>;
+		mobile?: Array<TokenObject>;
+	} = {
+		desktop: [],
+		mobile: []
+	};
 
-	const productMatches = content.matchAll(PRODUCT_WIDGET_REGEX);
-	const challengeMatches = content.matchAll(CHALLENGE_WIDGET_REGEX);
-	const sliderMatches = content.matchAll(SLIDER_WIDGET_REGEX);
-	const tagMatches = content.matchAll(TAG_WIDGET_REGEX);
-	const specificationMatches = content.matchAll(SPECIFICATION_WIDGET_REGEX);
-	const contactFormMatches = content.matchAll(CONTACTFORM_WIDGET_REGEX);
-	const pictureMatches = content.matchAll(PICTURE_WIDGET_REGEX);
-	const countdownMatches = content.matchAll(COUNTDOWN_WIDGET_REGEX);
-	const tagProductsMatches = content.matchAll(TAG_PRODUCTS_REGEX);
-	const galleryMatches = content.matchAll(GALLERY_WIDGET_REGEX);
+	const productMatches = (content + mobileContent).matchAll(PRODUCT_WIDGET_REGEX);
```

Comment:
```suggestion
	const productMatches = filterUndef([content, mobileContent]).map(text => text.matchAll(PRODUCT_WIDGET_REGEX)).flat(1);
```

Same for the others

More exact, especially for `orderedMatches`